lmnet/lmnet/networks/classification/vgg7.py

```diff
@@ -0,0 +1,268 @@
+            batch_size,
+            data_format,
+        )
+
+    def base(self, images, is_training=None):
+        keep_prob = tf.cond(is_training, lambda: tf.constant(0.5), lambda: tf.constant(1.0))
+        self.images = images
+
+        self.input = self.convert_rbg_to_bgr(images)
+        self.input = tf.cast(self.input, dtype=tf.float32)
+        self.input = tf.div(self.input, 255)
+
+        self.conv1 = self.conv_layer("conv1", self.input, filters=128, kernel_size=3, is_training=is_training)
+        self.conv2 = self.conv_layer("conv2", self.conv1, filters=128, kernel_size=3, is_training=is_training)
+
+        self.pool1 = self.max_pool("pool1", self.conv2, kernel_size=2, strides=2)
+
+        self.conv3 = self.conv_layer("conv3", self.pool1, filters=256, kernel_size=3, is_training=is_training)
+        self.conv4 = self.conv_layer("conv4", self.conv3, filters=256, kernel_size=3, is_training=is_training)
+
+        self.pool2 = self.max_pool("pool2", self.conv4, kernel_size=2, strides=2)
+
+        self.conv5 = self.conv_layer("conv5", self.pool2, filters=512, kernel_size=3, is_training=is_training)
+        self.conv6 = self.conv_layer("conv6", self.conv5, filters=512, kernel_size=3, is_training=is_training)
+
+        self.pool3 = self.max_pool("pool3", self.conv6, kernel_size=2, strides=2)
+
+        self.flatten = tf.contrib.layers.flatten(self.pool3)
+
+        self.fc1 = self.fc_layer("fc1", self.flatten, filters=1024, activation=tf.nn.relu)
+        self.fc1_drop = tf.nn.dropout(self.fc1, keep_prob)
+        self.fc2 = self.fc_layer("fc2", self.fc1_drop, filters=self.num_classes, activation=None)
+
+        return self.fc2
+
+    def conv_layer(
+        self,
+        name,
+        inputs,
+        filters,
+        kernel_size,
+        is_training=True,
+        strides=1,
+        padding="SAME",
+        activation=tf.nn.relu,
+        *args,
+        **kwargs
+    ):
+        kernel_initializer = tf.contrib.layers.xavier_initializer()
+        biases_initializer = tf.zeros_initializer()
+
+        if self.data_format == 'NHWC':
+            data_format = 'channels_last'
+        else:
+            data_format = 'channels_first'
+
+        #with tf.variable_scope(name, custom_getter=self.custom_getter):
+        conv = tf.layers.conv2d(inputs=inputs,
+                                filters=filters,
+                                kernel_size=kernel_size,
+                                kernel_initializer=kernel_initializer,
+                                #activation=None,
+                                #bias_initializer=biases_initializer,
+                                use_bias=False,
+                                padding=padding,
+                                strides=strides,
+                                data_format=data_format,
+                                kernel_regularizer=tf.contrib.layers.l2_regularizer(scale=0.0001),
+        )
+
+        batch_normed = tf.contrib.layers.batch_norm(conv,
```

Comment:
Author use caffe, the batch norm eps and decay is different.
http://caffe.berkeleyvision.org/tutorial/layers/batchnorm.html